packages/modular-scripts/src/addPackage.ts

```diff
@@ -16,82 +16,138 @@ import LineFilterOutStream from './utils/LineFilterOutStream';
 import { ModularPackageJson } from './utils/isModularType';
 
 const packagesRoot = 'packages';
+const CUSTOM_TEMPLATE = '__CHOOSE_MY_OWN__';
 
 interface AddOptions {
-  destination: string;
-  type: string | void;
   name: string | void;
+  type: string | void;
   template: string | void;
   preferOffline: boolean;
   verbose: boolean;
+  unstableName: string | void;
 }
 
-async function addPackage({
-  destination,
-  type: typeArg,
-  name: nameArg,
-  template: templateNameArg,
-  preferOffline = true,
-  verbose = false,
-}: AddOptions): Promise<void> {
-  const { type: packageType, name } =
-    (typeArg && nameArg ? { type: typeArg, name: nameArg } : null) ||
-    ((await prompts([
+async function promptForName(name: string | void) {
+  // The user may try to create a package using a name that already exists
+  // so we loop until they choose a unique name or cancel
+  while (true) {
+    // Default the response the predefined `name` argument
+    const response = name
+      ? { name }
+      : ((await prompts({
+          name: 'name',
+          type: 'text',
+          message: `What would you like to name this package?`,
+        })) as { name: string } | Record<string, never>);
```

Comment:
Why the `Record<string, never>`?

Reply:
If the user cancels the prompt with `Ctrl+C`, we get an empty object back, literally just `{}`. I've used `Record<string, never>` to represent that empty object and allow us to check `if (!response.name)` on line 44, covering both the case where the user enters no name by pressing enter or presses `Ctrl+C`.